### ISSUE TITLE
wasm-ld: error: cannot open output file hello_world.wasm: Input/output error

### ISSUE THREAD

**Issue by gushanke** [emscripten-core/emsdk#686]
Hi, My system is ubuntu18.04. My emsdk version is 2.0.12. When I execute "emcc hello_world.c -s WASM=1 -o hello_world.js" an error occurred.

**Comment by sbc100** [comment]
That is very odd.   It looks like some kind of fundamental issue with your filesystem.   What filesystem is mounted at the current working directory?   I seem to remember seeing these kinds of issues with qemu or other virtual envrioments when combined with windows filesystem.  Are you using some kind of virtual environment?

Can you run with `EMCC_DEBUG=1` and include the full output?

**Comment by gushanke** [comment]
> 这很奇怪。看来文件系统存在某种基本问题。当前工作目录中挂载了什么文件系统？我似乎还记得与Windows文件系统结合使用qemu或其他虚拟环境时遇到的这类问题。您是否正在使用某种虚拟环境？
> 
> 您可以运行`EMCC_DEBUG=1`并包含完整的输出吗？

Than you for your replay, Yes, I use the VmWare virtual machine. And Below is my result.
![46591afd-c3ec-4b81-8af5-a47e35de20ab](https://user-images.githubusercontent.com/45550209/105787943-6f40b700-5fba-11eb-80df-19ed58f83382.jpg)

**Comment by gushanke** [comment]
> That is very odd. It looks like some kind of fundamental issue with your filesystem. What filesystem is mounted at the current working directory? I seem to remember seeing these kinds of issues with qemu or other virtual envrioments when combined with windows filesystem. Are you using some kind of virtual environment?
> 
> Can you run with `EMCC_DEBUG=1` and include the full output?

Than you for your replay, Yes, I use the VmWare virtual machine. And Below is my result.
![001](https://user-images.githubusercontent.com/45550209/105788230-0ad22780-5fbb-11eb-8cf9-4d4e56a078a0.png)

**Comment by sbc100** [comment]
Yes, this seems to be some kind of bug in the way vmware exposes the windows filesystem to linux guest.

Can you use other llvm programs?  Specifically, I think the code for wasm-ld is the same as the normal lld.   Can you compile and link normal linux programs using llvm and lld?   

Short term fix: don't use /mnt/hgfs as the directory where you build to, but use a native linux filesystem.

**Comment by gushanke** [comment]
> Yes, this seems to be some kind of bug in the way vmware exposes the windows filesystem to linux guest.
> 
> Can you use other llvm programs? Specifically, I think the code for wasm-ld is the same as the normal lld. Can you compile and link normal linux programs using llvm and lld?
> 
> Short term fix: don't use /mnt/hgfs as the directory where you build to, but use a native linux filesystem.

Thank you very much, I will try it.

**Comment by sbc100** [comment]
Will be interesting to see if this is fixed by https://reviews.llvm.org/D81803

**Comment by andrewevstyukhin** [comment]
On the screenshot `-s` is missed before `EMCC_DEBUG=1`

**Comment by sbc100** [comment]
I believe this is/was an issue with linux VMs running on windows, not esmcripten/emsdk specific.
